File: leetCode/fukun/lc_053.js
const maxSubArray = function (nums) {
    let dp = nums[0];
    let op = nums[0];
    for (let i = 1; i < nums.length; i++) {
        dp = dp > 0 ? dp + nums[i] : nums[i];
        op = Math.max(op, dp);
    }
    return op;
};
console.log(maxSubArray([-2, 1, -3, 4, -1, 2, 1, -5, 4]));

File: leetCode/XSS/lc_053.java
import sun.plugin.liveconnect.OriginNotAllowedException;

public class lc_053 {
    public static void main(String[] args) {
//求连续最大子序和***
    }
    static class Solution {
        public int maxSubArray(int[] nums) {
            int ans = nums[0];
            int sum = 0;
            for(int num: nums) {
                if(sum > 0) {
                    sum += num;
                } else {
                    sum = num;
                }
                ans = Math.max(ans, sum);
            }
            return ans;
        }
    }
}
